src/app/components/UserTos.js

```diff
@@ -12,109 +13,110 @@ import Message from './Message';
+    setCheckedTos(!checkedTos);
+  };
 
-  handleSubmit() {
+  const handleSubmit = () => {
+    const onSuccess = () => {
+    };
     const onFailure = () => {
-      this.setState({
-        message: <FormattedMessage id="global.unknownError" defaultMessage="Sorry, an error occurred. Please try again and contact {supportEmail} if the condition persists." description="Message displayed in error notification when an operation fails unexpectedly" values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }} />,
-      });
+      setMessage(<FormattedMessage id="global.unknownError" defaultMessage="Sorry, an error occurred. Please try again and contact {supportEmail} if the condition persists." description="Message displayed in error notification when an operation fails unexpectedly" values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }} />);
     };
-
-    if (this.state.checkedTos) {
-      Relay.Store.commitUpdate(
-        new UpdateUserMutation({
-          current_user_id: this.props.user.id,
-          accept_terms: true,
-        }),
-        { onFailure },
-      );
-    }
-  }
-
-  handleValidate() {
-    if (!this.state.checkedTos) {
-      this.setState({
-        message: <FormattedMessage id="userTos.validation" defaultMessage="You must agree to the Terms of Service and Privacy Policy" description="Message to the user that they must review the application terms of service" />,
+    if (checkedTos) {
+      commitMutation(Relay.Store, {
+        mutation: graphql`
+          mutation UserTosUpdateUserMutation($input: UpdateUserInput!) {
+            updateUser(input: $input) {
+              me {
+                accepted_terms
+              }
+            }
+          }
+        `,
+        variables: {
+          input: {
+            id: props.user.id,
+            accept_terms: true,
+          },
+        },
+        onCompleted: onSuccess,
+        onError: onFailure,
       });
     }
-  }
+  };
 
-  render() {
-    const { user, about } = this.props;
+  const handleValidate = () => {
+    if (!checkedTos) {
+      setMessage(<FormattedMessage id="userTos.validation" defaultMessage="You must agree to the Terms of Service and Privacy Policy" description="Message to the user that they must review the application terms of service" />);
+    }
+  };
+  const { user, about } = props;
 
-    const actions = [
-      // eslint-disable-next-line jsx-a11y/click-events-have-key-events
-      <div onClick={this.handleValidate.bind(this)} style={{ cursor: 'pointer' }}>
-        <Button
-          id="tos__save"
-          color="primary"
-          onClick={this.handleSubmit.bind(this)}
-          disabled={!this.state.checkedTos}
-        >
-          <FormattedMessage id="userTos.save" defaultMessage="Save" description="Button label for the user to save their review of the terms of service" />
-        </Button>
-      </div>,
-    ];
+  const actions = [
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
```

Comment:
This is more of a flag for future work, but I would like to see these eslint exceptions removed from the code in the future. I think every onClick element should also accept an onKeyDown that checks for the "Enter" key being pressed. We could create a helper function that wraps existing functions, like

```jsx
<div onClick={handleClick} onKeyDown={callIfKeyDown(handleClick, 'Enter')} />
```

I'll file an issue for this. @caiosba